src/modules/thresholdsig/design.md

```diff
@@ -0,0 +1,134 @@
+# Status
+
+This PR has been open for a while and is blocked on progress on some
+supporting infrastructure; in particular the key setup protocol requires
+both a broadcast channel and private communication channels between
```

Comment:
I don't think the broadcast channel requirement should be viewed as a blocker for this PR because we can't solve that problem in this library. Assuming security of this scheme holds up, to get unforgeability it should be sufficient to make the implementer ensure that the pubcoeffs are the same for every participant. Obviously this is an enormous footgun.

We could add a function to compare pubcoeff hashes and signatures (independently). But this wouldn't solve the problem in general as it's probably easy to misuse and not straightforward to implement (how do you prevent replay?). Additionally, iit would be overkill for some situations. For example, sometimes there is that trusted party that can check the output on the displays of various hardware wallets match. Or a blockchain could be used as a broadcast channel.

The broadcast requirement at signing time is not as important because it's only DoS-preventing, so it could be safely offloaded to the implementer.

I agree that more in depth security analysis is a blocker.

Reply:
I'd say as long as we trust the broadcast channel only for robustness (= we terminate properly) and we do not trust it for unforgeability, this is not a blocker. And this should be true for this scheme. But yes, a security analysis should show that this is indeed the case.